frontend_lib/src/component/Tags/TagList.jsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import PropTypes from 'prop-types'
+import {
+  sendGlobalFlashMessage,
+  naturalCompare,
+  handleFetchResult
+} from '../../helper.js'
+import classnames from 'classnames'
+import NewTagForm from './NewTagForm.jsx'
+import Tag from './Tag.jsx'
+import {
+  getWorkspaceTagList,
+  getContentTagList,
+  deleteContentTag,
+  putContentTag
+} from '../../action.async.js'
+
+// require('./TagList.styl') // see https://github.com/tracim/tracim/issues/1156
+
+class TagList extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      tagList: []
+    }
+  }
+
+  toggleChecked = tagId => {
+    this.setState(previousState => {
+      const { props } = this
+      const checkedToggled = !previousState.tagIsChecked[tagId]
+      if (checkedToggled) {
+        putContentTag(props.apiUrl, props.workspaceId, props.contentId, tagId)
+      } else {
+        deleteContentTag(props.apiUrl, props.workspaceId, props.contentId, tagId)
+      }
+
+      return {
+        tagIsChecked: {
+          ...previousState.tagIsChecked,
+          [tagId]: checkedToggled
+        }
+      }
+    })
+  }
+
+  componentDidMount () {
+    this.updateTagList()
+  }
+
+  componentDidUpdate (prevProps) {
+    const { props } = this
+
+    if (prevProps.contentId !== props.contentId) {
+      this.updateTagList()
+    }
+  }
+
+  async updateTagList () {
+    const { props } = this
+    const fetchGetWsTagList = await handleFetchResult(
+      await getWorkspaceTagList(props.apiUrl, props.workspaceId)
+    )
+
+    if (!fetchGetWsTagList.apiResponse.ok) {
+      sendGlobalFlashMessage(props.t('Error while fetching a list of tags'))
+      return
+    }
+
+    const fetchGetContentTagList = await handleFetchResult(
+      await getContentTagList(props.apiUrl, props.workspaceId, props.contentId)
+    )
+
+    if (!fetchGetContentTagList.apiResponse.ok) {
+      sendGlobalFlashMessage(props.t('Error while fetching a list of tags'))
+    }
+
+    const isContentTag = (tag) => {
+      return fetchGetContentTagList.body.some(t => t.tag_id === tag.tag_id)
+    }
+
+    const sortTagList = (tagA, tagB) => {
+      const isTagAContent = isContentTag(tagA)
+      const isTagBContent = isContentTag(tagB)
+
+      if (!isTagAContent && isTagBContent) {
+        return 1
+      }
+
+      if (isTagAContent && !isTagBContent) {
+        return -1
+      }
+
+      return naturalCompare(tagA, tagB, props.i18n.language, 'tag_name')
+    }
+
+    const sortedTagList = fetchGetWsTagList.body.sort(sortTagList)
+
+    const tagIsChecked = {}
+
+    for (const tag of fetchGetContentTagList.body) {
+      tagIsChecked[tag.tag_id] = true
+    }
+
+    this.setState({ tagList: sortedTagList, tagIsChecked: tagIsChecked })
+  }
+
+  render () {
+    const { props, state } = this
+
+    return (
+      <div className='taglist' data-cy='taglist'>
```

Comment:
```suggestion
      <div className='tagList' data-cy='tagList'>
```
Just a suggestion, if you decide to apply, do not forget to update the rest of this file and the .styl one. Same for other words like `btnadd` = `btnAdd`

Reply:
done